docs/the-new-architecture/fabric-component-native-commands.md

```diff
@@ -0,0 +1,427 @@
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
+
+# Invoking native functions on your native component
+
+In the [base guide](/docs/fabric-native-components-introduction) to write a new Native Component, you have explored how to create a new component, how to pass properties from the JS side to the native side, and how to emit events from native side to JS.
+
+Custom components can also call some of the functions implementated in the native code imperatively, to achieve some more advanced functionalities, such as programmatically reload a web page.
+
+In this guide you'll learn how to achieve this, by using a new concept: Native Commands.
+
+This guide starts from the [Native Components](/docs/fabric-native-components-introduction) guide and assumes that you are familiar with it and that you are familiar with [Codegen](/docs/next/the-new-architecture/what-is-codegen).
+
+## 1. Update your component specs
+
+The first step is to update the component spec to declare the `NativeCommand`.
+
+<Tabs groupId="language" queryString defaultValue={constants.defaultJavaScriptSpecLanguage} values={constants.javaScriptSpecLanguages}>
+<TabItem value="typescript">
+
+Update the `WebViewNativeComponent.ts` as it follows:
+
+```diff title="Demo/specs/WebViewNativeComponent.ts"
+import type {HostComponent, ViewProps} from 'react-native';
+import type {BubblingEventHandler} from 'react-native/Libraries/Types/CodegenTypes';
+import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
++import codegenNativeCommands from 'react-native/Libraries/Utilities/codegenNativeCommands';
```

Comment:
also here: I think we should deep import from react-native

Reply:
we are deep importing from react-native. Codegen is the only exception on deep import.

Reply:
Sorry I meant **not** deep import. I believe those were recently fixed no?